### DISCUSSION TITLE
Corruption of in-memory data detected - XFS

### DISCUSSION THREAD

**Discussion by gr8matt** [Security-Onion-Solutions/securityonion#3203] [Unsupported Versions] [answered]
All,  Yesterday, I added a stick of RAM to my VMWare server that runs SO2 as well as a couple other VM's.  Everything went smoothly and all VM's were happy.  Yesterday afternoon, I had a hard power outage and my VMWare server went down unclean.  When I brought it back up, all other VM's are happy but Security Onion will not start up.  It gives me the following screen:

![image](https://user-images.githubusercontent.com/71660576/109528910-02fa1d00-7a6a-11eb-90c5-284868f025fd.png)

I am able to log on using my credentials to SO but am unable to do anything (including a ls).  

It does say to unmount the file system and run an xfs-repair but that command does not work.

Any ideas how I should proceed?  I will rebuild if I have to, I just didn't know if there was something that I am not seeing.

Thanks,
Matt

**Comment by weslambert** [comment, marked as answer]
I'd suggest rebuilding -- seems like something wrong with the underlying filesystem/disk

**Comment by KevCar87** [comment]
I too just ran into this. The fact that unmount is not available is troublesome. I know very little about Linux so I really don't know what to do. I was running this just to learn it and this happens. No backup so I guess I can re-install. Seems ridiculous not to have a step by step for this issue.

**Reply by Acewiza** [reply]
SO can be fairly disk-intensive in high-traffic environments, so if you pull the rug out from under it, all bets are off.  The array of possible ways things can go wrong will preclude any quick 'n easy fix-type solution in many cases, so a UPS is always encouraged, especially in unstable power locales and where up-time is deemed critical.

**Comment by isaacgolding** [comment]
Yea a rebuild "might" be your quickest fix.  If you are running a VM do you have a snapshot from a time prior to the crash?  If this is heavily tweaked install and you don't have a snapshot then you might want to look at using rescue options to attempt to recover the system.

XFS is a journaling filesystem which means that it basically writes down what is going to do in a special journal on the drive before it performs the actual filesystem operation.  What is supposed to happen is that if there is a crash, power failure etc any filesystem writes that happened just prior to or during the power outage are "re applied" when the system is brought back into normal operation.  But if the power died right when the journal was being written or the info is in a memory (ram) cache waiting to get into the journal (IO WAIT) then you can run into problems.   These errors don't "usually" bring the system to its knees.   But it does happen.     It's a lot more better than 10-15 years ago when there wasn't widespread use of journaled file systems.

It could be fixed... but would probably take more time than just reloading from snapshot/backup/fresh install.

Personally I'm a big fan of snapshots, one initial after everything is installed and setup and the system has been running for a day, and then another snapshot every week.   It makes it much easier when crashes occur, or I decide to go in and break something that was working perfectly fine before I touched it.

**Reply by KevCar87** [reply]
Appreciate the replies. I do not have a backup\snapshot. Been running it as a test and just trying to learn it. I have Alien Vault on the server as well. The power outage did not seem to affect it. Its not so much the re-install as its me wanting to know how to fix it. I don't use linux for the main reason of all the different filesystems and the different forks. Would like to learn it better though. I can boot into a linux rescue disk and see the other disks. Its just really confusing on what to do next. Examples given are also not much help as it always seems details are left out.

**Reply by KevCar87** [reply]
Update. I think I found the correct disk. I started a Centos Livecd. Found the disk that had a log file pending a replay. Mounted the drive. Its been mounting for about 20 minutes now. Assume its replaying the logfile. Will let it run and see what happens.

**Reply by KevCar87** [reply]
Well after a few hours I canceled it. Decided to run xfs_repair with the -L option. That was 20 hours ago. Still running. What a pain.

**Reply by KevCar87** [reply]
Gave up. Over 24 hours. No messages after running the command. Could see it was working but thats kind of ridiculous. I have had several linux servers running before and had power outages. All recovered or were easy to recover. But, this system is always writing. Guess I will start backing it up.

**Reply by btdewey** [reply]
We recently had VMWare snapshots on an ESXI server cause issues with security onion.  The storage space on the system was fairly tight and the snapshots put us in a position where there was very little available disk space on the host which caused slowing and instability.  With an NSM expect to almost double the hard drive storage with even just one snapshot.  We also noticed that queries ran faster once we removed the snapshots.  If your going to use snapshots on a heavily utilized search node ensure you have sufficient compute and storage resources.
